### ISSUE TITLE
enforceExtended and promisify not found

### ISSUE THREAD

**Issue by codechips** [ealush/vest#556]
Hi and thanks for the a great library!

I have trouble importing `enforceExtended` for some reason. Is the file missing or am I missing something? 

Things I've tried:

```js
// nope
import enforceExtended from 'vest/enforceExtented'; 

// installed n4s separately and tried this without success
import enforce from 'n4s/enforceExtended';
```
https://ealush.com/n4s/#/./business_rules

Is the documentation lagging behind? Should I install and use validator.js to solve email validation?

https://ealush.com/vest/#/./n4s/external

Same thing when trying to import `promisify` looks like.

```js
import promisify from 'vest/promisify';
// Package "vest" exists but package.json "exports" does not include entry for "./promisify".
```
I am using Snowpack as the bundler btw, but my gut feeling it's package related and not bundler specific.

Update: I can import promisify from `vest/promisify` so it might be bundler specific after all :thinking:

Also, this in Snowpack.

```js
// fails
import { test, enforce, create } from 'vest';
// [chrome]: The requested module '../../web_modules/vest.js' does not provide an export named 'test'
// [firefox]: Uncaught SyntaxError: import not found: enforce

// this works in both Chrome and FF
import vest from 'vest';
const { create, enforce, test } = vest;
 ```
Very strange. Any ideas? Is this maybe affected by how the final package is generated? I am just brainstorming here.

**Comment by ealush** [comment]
Hey @codechips, thank you for reaching out, and sorry for the delayed response. I completely missed this issue.
🙏🏻

### Promisify
As you noted, this seems to be a bundler thing. I tested Vest with node imports/webpack/rollup/parcel, never tried it with snowpack or browser imports. I'll have some reading on it and see what needs to be changed. Thank you for reporting that. 

Btw, trying to import promisify directly seems to be fine on my end as well: https://codesandbox.io/s/vigilant-forest-nse05?file=/src/index.js

---

### enforceExtended
In vest/enforce v2 I added enforceExtended as a wrapper around some common validatorJS functions, thinking it would reduce the cognitive load of adding multiple packages for validation, but that came with a cost of bloat and needlessly increased bundle size.

enforceExtended used to be ~29KB, while validatorjs `isEmail` itself is about 3KB. The difference is due to the fact that enforceExtended contained multiple functions that one might not actually need in a given project. So when publishing v3 I decided to drop it, and I recommend everyone who asks to use whatever it is you need from validatorjs directly, because they are just awesome, and are the best at what they do. In fact, from my research, it seems that most other packages use them internally as well (and pay the bundle size price...).

Seems like there's a CI problem that stopped the n4s docs site from updating along with this repo, I'll take care of it. Sorry that this lead to so much confusion.

I'm keeping this one open for a while and I'll close it down if I don't see this issue repeating.

**Comment by codechips** [comment]
No problem, thanks for the answer. Ping me if you need a simple Snowpack project in order to reproduce the issue. I am using Svelte, but I doubt the framework is the issue here.

**Comment by ealush** [comment]
Yes, a boilerplate project would be amazing for me to debug these issues. I'd love that.

**Comment by codechips** [comment]
Here is a simple Snowpack project that demonstrates the problem https://github.com/codechips/svelte-snowpack-vest-bug

**Comment by ealush** [comment]
Thank you @codechips, on it. I'll report back with the findings.

**Comment by codechips** [comment]
I forgot to mention one other thing. When importing Promisify I have to explicitly specify `.js` extension to make it work.
```js
// doesn't work
import promisify from 'vest/promisify';
// works
import promisify from 'vest/promisify.js';
```

**Comment by ealush** [comment]
OK, so first things first - I created a version that on my end seem to resolve the named imports issue. Can you confirm that it works for you as well?
```vest@3.1.2-dev-6a05bd```
This is the version number. It is a development release that I created specifically for this issue. If it works correctly I'll ship it with the next version.

I'll now try to figure out the promisify.js thing.

**Comment by ealush** [comment]
@codechips, ok, so I created a new version that seems to fix the named imports on my end. Can you please try it out and let me know if it solves it for you as well? If it does, I'll publish it with the next version. 
For now this build is on: `vest@3.1.2-dev-6a05bd`.

I'll now look into the promisify.js thing.

Thank you!

**Comment by codechips** [comment]
Checked and it works! Nice! :+1:

**Comment by ealush** [comment]
@codechips 
I seem to have found the reason that you're unable to import promisify without the extension. I made another dev version with the fix applied. Can you please test again that it works on your end (both with and without the extension).

```vest@3.1.2-dev-a2bd9c```

Thank you for reporting these issues, making vest better!

**Comment by codechips** [comment]
Tried and everything now works as expected! Thanks! Closing this.

**Comment by ealush** [comment]
@codechips, I just released 3.1.2 that includes both these fixes. Thank you for your patience.
https://github.com/ealush/vest/releases/tag/3.1.2
